### DISCUSSION TITLE
Allow rgw_enable_usage_log to be set to false

### DISCUSSION THREAD

**Discussion by ushitora-anqou** [rook/rook#14737] [Q&A] [answered]
In the current Rook implementation, `rgw_enable_usage_log` is set to `true`.

https://github.com/rook/rook/blob/d4f3bf3f2cff7b30738eef9e9e4fab844376f773/pkg/operator/ceph/object/config.go#L168

Because of this setting, usage logs can take up a lot of storage space. So, I'd like to set this field to `false`, but it seems that Rook doesn't provide any way to change this value.

Is there any workaround for this? And is there any reason for the current behavior?

**Comment by ushitora-anqou** [comment]
Sorry, I totally forgot that we can inject any Ceph configuration using a ConfigMap, so there’s a workaround.

So, my precise question is: Is there any reason why Rook changes the `rgw_enable_usage_log` value from `false` (which is Ceph's default) to `true`?

**Reply by travisn** [reply]
That option was set a long time ago in #2738, not sure why.

Does setting that config override actually work? IIRC the `ceph config set` options will override any settings in the configmap, so I'm not sure that will work.

**Reply by ushitora-anqou** [reply]
@BlaineEXE Do you remember why this change was made? If there's no practical reason, I'd like to change the value to false.

**Reply by ushitora-anqou** [reply]
@travisn Thank you for your reply.

> Does setting that config override actually work? IIRC the ceph config set options will override any settings in the configmap, so I'm not sure that will work.

And this seems correct. I just tried the "workaround" I suggested and found that it didn't work :(

**Reply by BlaineEXE** [reply]
Those settings were pre-existing even before https://github.com/rook/rook/pull/2738. I think that PR only changed how they get set. (previously in `pkg/daemon/ceph/rgw/init.go`, line 105) Setting the desired config in the `rook-config-override` configmap should override the operator setting.

If your workaround used a different configmap, please use `rook-config-override` and try again. Otherwise, if you post what you added to the config, we might be able to debug why that didn't work.

**Reply by BlaineEXE** [reply]
That config is sufficiently old that I'm not sure how to figure out why it was added. 

I wonder if this config has any consideration alongside this feature request that @ideepika has been looking at: https://github.com/rook/rook/issues/14372

**Reply by travisn** [reply]
IIRC Ceph's precedence (highest to lowest) are CLI options, ceph config, then ceph.conf. The configmap settings translate to ceph.conf, so I don't see a way to override that without a code change.

**Reply by ushitora-anqou** [reply]
@BlaineEXE @travisn I've confirmed that `rook-config-override` can't change the `rgw_enable_usage_log` value using [this code](https://github.com/ushitora-anqou/rook-on-minikube/tree/rook-14737). I'd appreciate it if you could let me know if I've done anything wrong.

**Reply by BlaineEXE** [reply]
@travisn that isn't right. It's CLI, env, configfile (ceph.conf), then mon config database (ceph config), then builtin defaults. The `rook-config-override` (ceph.conf) should override any mon configs. Only env vars and CLI flags can override the config file.

`ceph config dump` only shows the configs in the mon database and not the configs for the daemons themselves, which is why you aren't seeing what's configured in the config file. Use `ceph config show {daemon-type}.{id} ` to see daemon configs to try to verify the configs specifically for the RGW.

All the info I provided here came from this ceph doc page: https://docs.ceph.com/en/reef/rados/configuration/ceph-conf/

**Reply by travisn** [reply, marked as answer]
Thanks for the clarification on precedence between ceph.conf and mon store. In any case, since rook is setting this as a CLI option, there is no way to override it currently.

**Reply by ushitora-anqou** [reply]
Thank you for the information. I just confirmed that, even when using `ceph config show`, the value can't be overwritten by `rook-config-override`. I updated the README of my code above to reflect this.

I had two main questions, and I think I found the answers:

- There seem to be no clear reasons why Rook sets `rgw_enable_usage_log` to `true` by default.
- Currently, there's no workaround to avoid this behavior in Rook.

I’m going to share these findings with my team so that we can decide our next steps. Thanks a lot for your help!

**Reply by travisn** [reply]
I'd suggest opening an [issue](https://github.com/rook/rook/issues) that allows overriding any of the [rgw settings](https://github.com/rook/rook/blob/0481c8f64a6bc48a23c94c0b1d609c9d152ed893/pkg/operator/ceph/object/config.go#L166-L170) that we are setting by default currently.
